I get the default Helix behavior instead of evil-helix behavior

### Summary

Let's continue the discussion from Reddit.

It seems I totally get the behavior of Helix instead of evil-helix.

Doing `v` shows `SEL` and I have the purple color scheme by default.

I compiled with `cargo build --release` and I run with `./target/release/hx`.
I also tried `cargo install --path helix-term --locked` which seems the way to build `helix` from source and `which hx` returns `/home/user/.cargo/bin/hx`.
`hx --version` returns `helix 24.3 (d969d507)`.

Here's the start of the output of `hx --health`:
```
Config file: default
Language file: default
Log file: /home/user/.cache/helix/helix.log
Runtime directories: /home/user/.config/helix/runtime;/home/user/.cargo/bin/runtime
Runtime directory does not exist: /home/user/.config/helix/runtime
Runtime directory does not exist: /home/user/.cargo/bin/runtime
Clipboard provider: xclip
System clipboard provider: xclip
```

### Reproduction Steps

I tried this:

1. `cargo run --release`

I expected this to happen:
Have vim keybindings.

Instead, this happened:
I have the helix keybindings and the purple background.

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2023-05-23T08:14:16.620 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-05-23T12:15:55.583 helix_lsp::transport [ERROR] err <- "[ERROR rust_analyzer::main_loop] flycheck 0: File with cargo diagnostic not found in VFS: file not found: /home/user/.rustup/toolchains/nightly-2023-03-02-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/macros/mod.rs\n"
2023-05-23T12:15:55.583 helix_lsp::transport [ERROR] err <- "[ERROR rust_analyzer::main_loop] flycheck 0: File with cargo diagnostic not found in VFS: file not found: /home/user/.rustup/toolchains/nightly-2023-03-02-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/macros/mod.rs\n"
2023-05-23T12:16:04.078 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
```

</details>


### Platform

Linux

### Terminal Emulator

Alacritty

### Installation Method

From source

### Helix Version

helix 24.3 (d969d507)

--- a/helix-term/src/config.rs
+++ b/helix-term/src/config.rs
@@ -29,8 +29,8 @@ impl Default for Config {
     fn default() -> Config {
         Config {
             theme: None,
-            keys: keymap::default(),
-            editor: helix_view::editor::Config::default(),
+            keys: keymap::default_evil(),
+            editor: helix_view::editor::Config::default_evil(),
         }
     }
 }

--- a/helix-view/src/editor.rs
+++ b/helix-view/src/editor.rs
@@ -877,7 +877,7 @@ pub enum PopupBorderConfig {
 impl Default for Config {
     fn default() -> Self {
         Self {
-            evil: true,
+            evil: false,
             scrolloff: 5,
             scroll_lines: 3,
             mouse: true,
@@ -927,6 +927,15 @@ impl Default for Config {
             indent_heuristic: IndentationHeuristic::default(),
             jump_label_alphabet: ('a'..='z').collect(),
         }
+    }
+}
+
+impl Config {
+    pub fn default_evil() -> Self {
+        let mut config = Config::default();
+        config.evil = true;
+        config.statusline.mode = ModeConfig::default_evil();
+        return config;
     }
 }
 

